src/umleditor/mvc_model/entity.py

```diff
@@ -215,6 +215,74 @@ def set_method_name(self, new_name):
         """
         self._name = new_name
 
+    def add_parameters(self, params: list[str]):
+        """
+        Adds a list of new parameters to the method.
+
+        Args:
+            params (list[str]): The list of new parameters to be added.
+
+        Raises:
+            CustomExceptions.ParameterExistsError: If any of the parameter already exists in the method.
+
+        Returns:
+            None.
+        """
+        for new_param in params:
+            if new_param in self._params:
+                raise CustomExceptions.ParameterExistsError(new_param)
+        self._params.extend(params)
+
+    def remove_parameters(self, params: list[str]):
+        """
+        Removes a list of parameters from the method.
+
+        Args:
+            params (list[str]): The list of parameters to be removed.
+
+        Raises:
+            CustomExceptions.ParameterNotFoundError: If any of the parameter does not exist in the method.
+
+        Returns:
+            None.
+        """
+        for remove_param in params:
+            if remove_param not in self._params:
+                raise CustomExceptions.ParameterNotFoundError(remove_param)
+        for remove_param in params:
+            del self._params[self._params.index(remove_param)]
+
+    def change_parameters(self, old_params: list[str], new_params: list[str]):
```

Comment:
I don't think this method needs to be able to replace a partial list. It assumes that if it is replacing params it replaces the existing list with a new list.

Reply:
I asked Dr. Schwartz.